compose/automation/expr_types.go

```diff
@@ -171,26 +174,61 @@ func CastToComposeRecordValues(val interface{}) (out types.RecordValueSet, err e
 	switch val := expr.UntypedValue(val).(type) {
 	case *types.Record:
 		return val.Values, nil
+	case *types.RecordValue:
+		return types.RecordValueSet{val}, nil
 	case types.RecordValueSet:
 		return val, nil
 	case map[string]string:
-		out = types.RecordValueSet{}
-		for k, v := range val {
-			out = out.Set(&types.RecordValue{Name: k, Value: v})
+		for _, k := range slice.Keys(val) {
+			out = out.Set(&types.RecordValue{Name: k, Value: val[k]})
 		}
-
 		return
-
 	case map[string][]string:
-		out = types.RecordValueSet{}
-		for k, vv := range val {
-			for i, v := range vv {
+		for _, k := range slice.Keys(val) {
+			for i, v := range val[k] {
 				out = out.Set(&types.RecordValue{Name: k, Value: v, Place: uint(i)})
 			}
 		}
-
 		return
+	case map[string]interface{}:
+		for _, k := range slice.Keys(val) {
+			if val[k] == nil {
+				continue
+			}
+
+			var vv []interface{}
+			// covering a couple of typed slices
+			ref := reflect.ValueOf(val[k])
+
+			if ref.Kind() == reflect.Slice {
+				for i := 0; i < ref.Len(); i++ {
+					vv = append(vv, ref.Index(i).Interface())
+				}
+			} else {
+				vv = []interface{}{val[k]}
+			}
+
+			i := 0
+			for _, value := range vv {
+				switch v := value.(type) {
+				case bool:
+					if !v {
+						// a small exception for boolean fields, don't add false values
+						continue
+					} else {
+						// when true, cast to 1
+						// legacy ...
+						value = "1"
+					}
```

Comment:
I would prefer to see these comments explaining the booleans below the `case bool:` line; all together